### ISSUE TITLE
Integration Flow strange behaviour

### ISSUE THREAD

**Issue by marctud** [spring-projects/spring-integration-java-dsl#139]
Hello,

I have a problem with an integration flow.
````java
return IntegrationFlows.from(Amqp.inboundGateway(simpleListnerContainer))
           .transform(Transformers.fromJson(ClassToTransform.class))
           .handle(service)
           .transform(Transformers.toJson())
           .log()
           .get();
````

Because of that `.log()` this integration flow fails with `Dispatcher has no subscribers`:

```
Caused by: org.springframework.integration.MessageDispatchingException: Dispatcher has no subscribers
	at org.springframework.integration.dispatcher.UnicastingDispatcher.doDispatch(UnicastingDispatcher.java:154)
	at org.springframework.integration.dispatcher.UnicastingDispatcher.dispatch(UnicastingDispatcher.java:121)
	at org.springframework.integration.channel.AbstractSubscribableChannel.doSend(AbstractSubscribableChannel.java:89)
	... 79 common frames omitted
```

If I comment that `.log()` line, everything works fine.
Can someone help me understand why is this?
Thank you

**Comment by artembilan** [comment]
That has been fixed in the Spring Integration 5.0 recently: https://jira.spring.io/browse/INT-4210.
And see explanation and solution in the StackOverflow: https://stackoverflow.com/questions/41695292/httprequestexecutingmessagehandler-no-output-channel-or-replychannel-header-ava/41699176#41699176

Closed as fixed.

**Comment by marctud** [comment]
Thank you so much for the fast reply :)
